pkg/vsphere/tasks/waiter.go

```diff
@@ -27,12 +28,40 @@ import (
 	"github.com/vmware/govmomi/vim25/progress"
 	"github.com/vmware/govmomi/vim25/soap"
 	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/pkg/trace"
 )
 
 const (
 	maxBackoffFactor = int64(16)
+
+	VMContextObjectKey = "VMContextObject"
+)
+
+var (
+	m             sync.RWMutex
+	errorHandlers []func(ctx context.Context, err error) (bool, error)
```

Comment:
Also, does the handler need to have additional context (not the go package), i.e. additional arguments? Like maybe task info, or the object that was involved in the task?